spec/rspec/support/differ_spec.rb

```diff
@@ -261,6 +261,35 @@ def inspect; "<BrokenObject>"; end
           expect(diff).to be_diffed_as(expected_diff)
         end
 
+        it "outputs unified diff message of two hashes with BigDecimal keys" do
+          in_sub_process_if_possible do
+            require 'bigdecimal'
+            expect {
+              diff = differ.diff({ BigDecimal('3') => 'b'}, { 3 => 'b' })
+              if expect(diff).to match(/3.0 \(#<BigDecimal/)
+                raise StandardError, 'Passed Test'
+              end
+            }.to raise_error(StandardError, 'Passed Test')
+          end
+        end
+
+        context 'with ActiveSupport' do
```

Comment:
The differ knows nothing about activesupport and it doesn't seem appropriate to specifically call that out and test it here.  Isn't the behavioral change simply that it uses the `ObjectInspector` to format hashes?  Maybe we can do one example using `Time` objects in hashes (which were the original motivation for special formatting) rather than one BigDecimal and one AS example.